cli/login.go

```diff
@@ -119,7 +120,26 @@ func LoginCommand(app *kingpin.Application, input LoginCommandInput) {
 		return
 	}
 
-	req, err := http.NewRequest("GET", "https://signin.aws.amazon.com/federation", nil)
+	loginUrlPrefix := "https://signin.aws.amazon.com/federation"
+	destination := "https://console.aws.amazon.com/"
+
+	if profile, _ := awsConfig.Profile(input.Profile); profile.Region != "" {
+		destinationDomain := "console.aws.amazon.com"
+		switch {
+		case strings.HasPrefix(profile.Region, "cn-"):
+			loginUrlPrefix = "https://signin.amazonaws.cn/federation"
+			destinationDomain = "console.amazonaws.cn"
+		case strings.HasPrefix(profile.Region, "us-gov-"):
+			loginUrlPrefix = "https://signin.amazonaws-us-gov.com/federation"
+			destinationDomain = "console.amazonaws-us-gov.com"
+		}
+		destination = fmt.Sprintf(
+			"https://%s.%s/console/home?region=%s",
+			profile.Region, destinationDomain, profile.Region,
+		)
+	}
```

Comment:
Could we pull this logic out into its own function?

Reply:
That sounds reasonable. It'd certainly make it cleaner. I'll get on it.